src/components/UI/GridTable/GridCheckBox.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  Checkbox,
+  FormControlLabel,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@mui/material';
+import { castGridStyles } from './GridTable';
+
+interface GridTableProps {
+  entities: string[];
+  selectedEntities: string[];
+  onSelect: (selected: string[]) => void;
+  title: string;
+}
+
+export const GridTableCheckBox = ({
+  title,
+  entities,
+  selectedEntities,
+  onSelect,
+}: GridTableProps) => {
+  const handleSelect = (column: string) => {
+    if (selectedEntities.includes(column)) {
+      onSelect(selectedEntities.filter((col) => col !== column));
+    } else {
+      onSelect([...selectedEntities, column]);
+    }
+  };
+
+  const handleSelectAll = () => {
+    if (selectedEntities.length === entities.length) {
+      onSelect([]);
+    } else {
+      onSelect(entities);
+    }
+  };
+
+  return (
+    <Table sx={castGridStyles.container}>
+      <TableHead sx={castGridStyles.headerItem}>
+        <TableRow>
+          <TableCell
+            sx={{
+              padding: '12px 10px',
+              width: `100%`,
+              borderRight: '1px solid #E8E8E8',
+              display: 'flex',
+              alignItems: 'center',
+            }}
```

Comment:
The table is common in GridTable and GridCheckbox. Can we extend the GridTable component to create GridCheckbox. This way if some style changes are needed we don't have to update in two places

Reply:
yes agreed. Let me think about it and make the necessary changes.

Reply:
Done. Have re-used the gridtable